src/D1_model_components.py

```diff
@@ -399,65 +399,61 @@ def source_non_dispatchable_optimize(model, dict_asset, **kwargs):
+            # and add it to solph.Flow()
+            if "maximumCap" in dict_asset and dict_asset["maximumCap"]["value"] > 0:
                 outputs[kwargs["busses"][bus]] = solph.Flow(
                     label=dict_asset["label"],
-                    max=dict_asset["timeseries_normalized"],
+                    actual_value=dict_asset["timeseries_normalized"],
+                    fixed=True,
+                    existing=dict_asset["installedCap"]["value"],
+                    investment=solph.Investment(
+                        ep_costs=dict_asset["simulation_annuity"]["value"]
+                        / dict_asset["timeseries_peak"]["value"],
+                        maximum=dict_asset["maximumCap"]["value"],
+                    ),
+                    variable_costs=dict_asset["opex_var"]["value"][index]
+                    / dict_asset["timeseries_peak"]["value"],
+                )
+            else:
+                outputs[kwargs["busses"][bus]] = solph.Flow(
+                    label=dict_asset["label"],
+                    actual_value=dict_asset["timeseries_normalized"],
+                    fixed=True,
+                    existing=dict_asset["installedCap"]["value"],
                     investment=solph.Investment(
                         ep_costs=dict_asset["simulation_annuity"]["value"]
                         / dict_asset["timeseries_peak"]["value"]
                     ),
-                    variable_costs=dict_asset["opex_var"]["value"][0]
+                    variable_costs=dict_asset["opex_var"]["value"][index]
                     / dict_asset["timeseries_peak"]["value"],
                 )
                 index += 1
+
+    else:
+        if "maximumCap" in dict_asset and dict_asset["maximumCap"]["value"] > 0:
+            outputs = {
+                kwargs["busses"][dict_asset["output_bus_name"]]: solph.Flow(
+                    label=dict_asset["label"],
+                    actual_value=dict_asset["timeseries_normalized"],
+                    fixed=True,
+                    existing=dict_asset["installedCap"]["value"],
+                    investment=solph.Investment(
+                        ep_costs=dict_asset["simulation_annuity"]["value"]
+                        / dict_asset["timeseries_peak"]["value"],
+                        maximum=dict_asset["maximumCap"]["value"],
+                    ),
+                    variable_costs=dict_asset["opex_var"]["value"]
+                    / dict_asset["timeseries_peak"]["value"],
+                )
+            }
         else:
             outputs = {
                 kwargs["busses"][dict_asset["output_bus_name"]]: solph.Flow(
                     label=dict_asset["label"],
-                    max=dict_asset["timeseries_normalized"],
+                    actual_value=dict_asset["timeseries_normalized"],
+                    fixed=True,
+                    existing=dict_asset["installedCap"]["value"],
```

Comment:
What did you do here? In the github comparison thing a solph.Flow with max is replaced by a solph.flow with actual value, fixed and existing cap - it might be though that there is a line mix-up.
It does seem correct for a non-dispatchable source.